tests/webapp/api/test_performance_data_api.py

```diff
@@ -235,3 +235,134 @@ def test_filter_data_by_framework(webapp, test_repository, test_perf_signature,
     assert len(datums) == 1
     assert datums[0]['job_id'] == 1
     assert datums[0]['signature_id'] == 2
+
+
+def test_filter_signatures_by_interval(webapp, test_repository, test_perf_signature):
+    # interval for the last 24 hours, only one signature exists last updated within that timeframe
+    resp = webapp.get(reverse('performance-signatures-list',
+                              kwargs={
+                                  "project": test_perf_signature.repository.name
+                              }) + '?interval={}'.format(86400))
+    assert resp.status_int == 200
+    assert len(resp.json.keys()) == 1
+    assert resp.json[test_perf_signature.signature_hash]['id'] == 1
+
+
+def test_filter_signatures_by_range(webapp, test_repository, test_perf_signature):
+    today = datetime.datetime.now()
+    one_day_ago = today - datetime.timedelta(days=1)
+    three_days_ago = today - datetime.timedelta(days=3)
+    seven_days_ago = today - datetime.timedelta(days=7)
+
+    # set signature last updated to 3 days ago
+    test_perf_signature.last_updated = three_days_ago
+    test_perf_signature.save()
+
+    # start date 3 days ago, should find 1 signature
+    resp = webapp.get(reverse('performance-signatures-list',
+                              kwargs={
+                                  "project": test_perf_signature.repository.name
+                              }) + '?start_date={}'.format(three_days_ago))
+    assert resp.status_int == 200
+    assert len(resp.json.keys()) == 1
+    assert resp.json[test_perf_signature.signature_hash]['id'] == 1
+
+    # start date one day ago, so won't find any signatures
+    resp = webapp.get(reverse('performance-signatures-list',
+                              kwargs={
+                                  "project": test_perf_signature.repository.name
+                              }) + '?start_date={}'.format(one_day_ago))
+    assert resp.status_int == 200
+    assert len(resp.json.keys()) == 0
+
+    # start date 7 days ago, end date 1 day ago, should find 1 signature
+    resp = webapp.get(reverse('performance-signatures-list',
+                              kwargs={
+                                  "project": test_perf_signature.repository.name
+                              }) + '?start_date={}&end_date={}'.format(seven_days_ago, one_day_ago))
+    assert resp.status_int == 200
+    assert len(resp.json.keys()) == 1
+    assert resp.json[test_perf_signature.signature_hash]['id'] == 1
```

Comment:
For this sort of unit test case, it can be helpful for clarity to parametrize the test so you iterate through the different cases (with different expected results). Could you do that here (and below)? Here's an example:

https://github.com/mozilla/treeherder/blob/master/tests/webapp/api/test_logslice_api.py#L12